src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,247 @@
+import { usePopper } from 'react-popper';
+
+import { theme } from '../../essentials/theme';
+import { get } from '../../utils/themeGet';
+import { Colors, Spaces } from '../../essentials';
+import { ChevronDownIcon, ChevronUpIcon } from '../../icons/index';
+import { useClickOutside } from '../../utils/hooks/useClickOutside';
+
+import { Text } from '../Text/Text';
+
+import { PopoverContent } from './PopoverContent';
+
+interface PopoverRefObjectProps {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ref: any;
+}
+
+const PopoverTrigger = styled.div.attrs({ theme })<PopoverRefObjectProps>`
+    display: inline-block;
+    width: fit-content;
+    border-radius: ${get('radii.2')};
+`;
+
+const DefaultPopoverWrapper = styled.div.attrs({ theme })`
+    position: relative;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    border: 1px solid ${get('semanticColors.button.secondary.borderHover')};
+    padding: 0.8125rem ${Spaces[2]};
+    border-radius: ${get('radii.2')};
+
+    &:hover {
+        cursor: pointer;
+        background-color: ${get('semanticColors.background.secondary')} !important;
+    }
+`;
+
+const PopoverContentContainer = styled.div<PopoverRefObjectProps>`
+    display: inline-block;
+`;
+
+const PopoverContentWrapper = styled.div`
+    display: inline-block;
+    box-sizing: border-box;
+    width: auto;
+    height: auto;
+    z-index: 1000;
+    box-shadow: ${get('shadows.small')};
+    max-height: none;
+
+    &:focus {
+        outline: 0;
+    }
+`;
+
+const KEY_CODE_MAP = {
+    ENTER: 13,
+    SPACE: 32,
+    ESC: 27
+};
+
+interface PopoverProps {
+    /**
+     * Popover Trigger (Only use Text, Link, Button or Icon component from @wave as a trigger)
+     */
+    children: React.ReactNode;
+    /**
+     * Popover content (can be any valid React Element or component containing React Elements)
+     */
+    content: React.ReactNode;
+    /**
+     * Optional: Specify the Popover content placement (placement changes automatically, if Popover content cannot fit inside viewport with selected placement)
+     */
+    placement?: Placement;
+    /**
+     * Optional: Specify the Popover content offset (margin between Popover trigger and content)
+     */
+    offset?: number;
+    /**
+     * Optional: Render popover content open by default
+     */
+    isOpen?: boolean;
+    /**
+     * Optional: Define a callback for when Popover content is opened
+     */
+    onOpen?: () => void;
+    /**
+     * Optional: Define a callback for when Popover content is closed
+     */
+    onClose?: () => void;
+}
+
+export const Popover: React.FC<PopoverProps> = ({
+    children,
+    content = '',
+    placement = 'bottom-start',
+    offset = 5,
+    isOpen = false,
+    onOpen,
+    onClose
+}: PopoverProps) => {
+    const [triggerReference, setTriggerReference] = React.useState(undefined);
+    const [popperReference, setPopperReference] = React.useState(undefined);
```

Comment:
Can we call this `contentReference` instead? I find it easier to understand if we use the same terms as in our docs